### PR TITLE
Add pygram11 recipe

### PR DESCRIPTION
**Pull request by drdavis**
Adding recipe for pygram11, a small python histogramming library built on pybind11 and accelerated with OpenMP.

### CODE REVIEW

**Comment by drdavis** [comment]
The package will build without OpenMP, so the build is passing. However, I'm wondering why OpenMP is not being found on the `osx` build, I'm ensuring `openmp` is required for `osx` in `requirements.host` and `requirements.run`.

cc @conda-forge/help-c-cpp

**Comment by drdavis** [comment]
OpenMP headers are discovered but the `-fopenmp -Xprepreprocessr` isn't used and neither is `lomp`

**Comment by drdavis** [comment]
OK it looks like the library discovery worked by adding `openmp` to `requirements.build` but now the library isn't loaded
```
dyld: Library not loaded: @rpath/libomp.dylib
```

**Comment by drdavis** [comment]
@beckermr the documentation mentions you as a contact person for help with OpenMP on macOS; any insights? Thanks!

**Comment by beckermr** [comment]
Yeah this is fun. Add llvm-openmp to the build, host and run sections. I need to actually make a pr on those docs.

**Comment by beckermr** [comment]
Bahahaha. You had that. Well then idk what to do.

**Comment by drdavis** [comment]
Oh well 😄OpenMP works on Linux -- and on macOS OpenMP can work with a pip installation, so I'm satisfied with this for now.

**Comment by beckermr** [comment]
This line doesn't appear to set the rpath correctly

```
x86_64-apple-darwin13.4.0-clang -DNDEBUG -fwrapv -O3 -Wall -Wstrict-prototypes -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -I$PREFIX/include -fdebug-prefix-map=${SRC_DIR}=/usr/local/src/conda/${PKG_NAME}-${PKG_VERSION} -fdebug-prefix-map=${PREFIX}=/usr/local/src/conda-prefix -D_FORTIFY_SOURCE=2 -mmacosx-version-min=10.9 -I/usr/local/include -c test_openmp.c -o objects/test_openmp.o -Xpreprocessor -fopenmp
```

This is probably why the linker cannot find the dylib. It is definitely there.

**Comment by drdavis** [comment]
OK so this needs `-Wl,-rpath,<path-to-omp>`

**Comment by drdavis** [comment]
Success 👍 added `-Wl,-rpath,$PREFIX/lib` to the linker. Thanks for pointing me to that

File: recipes/pygram11/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  host:
+    - llvm-openmp  # [osx]
+    - python
+    - pip
+    - setuptools >=39.0.1
+    - pybind11 >=2.2
+    - numpy >=1.12
+  run:
+    - llvm-openmp  # [osx]
+    - numpy >=1.12
```
Comment (beckermr) [inline recipes/pygram11/meta.yaml]: See the docs for how to build against numpy with a version constraint. 

https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

**Comment by drdavis** [comment]
> Doulble check the test env. You need to use 1.12.* I think too.

**Comment by drdavis** [comment]
Whoops didn't mean to close...
```
requirements:
  hist:
    - numpy 1.12.*
```

crashed in `./.circleci/run_docker_build.sh` on my machine

**Comment by beckermr** [comment]
Ahhhh ok. The docs are out of date then. What is the output version constraint at runtime for the working version?

**Comment by drdavis** [comment]
I'm not sure I know what you mean, are you talking about the numpy version constraint or the pygram11 version?

**Comment by beckermr** [comment]
So if you open up the output package, you can checkout the dependence of the final package. We should make sure the final runtime requirement for numpy is not restricted to only the 1.12.XYZ series.

**Comment by drdavis** [comment]
Looks like numpy 1.16.2 is being used for the tests, so we're good on not being restricted.

**Comment by beckermr** [comment]
Ohhh that looks correct actually. My knowledge of conda is outdated. LGTM!

**Pull request merged**